google/cloud/aiplatform/datasets.py

```diff
@@ -15,6 +15,358 @@
+from google.cloud.aiplatform_v1beta1 import ImportDataConfig
+from google.cloud.aiplatform_v1beta1 import ImportDataRequest
+from google.cloud.aiplatform_v1beta1 import Dataset as GapicDataset
+from google.cloud.aiplatform_v1beta1 import DatasetServiceClient
+
+
+# TODO(b/171275584): Add async support to Dataset class
+class Dataset(base.AiPlatformResourceNoun):
+    """Managed dataset resource for AI Platform"""
+
+    client_class = DatasetServiceClient
+    _is_client_prediction_client = False
+
+    def __init__(
+        self,
+        dataset_name: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves an existing managed dataset given a dataset name or ID.
+
+        Args:
+            dataset_name (str):
+                Required. A fully-qualified dataset resource name or dataset ID.
+                Example: "projects/123/locations/us-central1/datasets/456" or
+                "456" when project and location are initialized or passed.
+            project (str):
+                Optional project to retrieve dataset from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional location to retrieve dataset from. If not set, location
+                set in aiplatform.init will be used.
+            credentials: Optional[auth_credentials.Credentials]=None,
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        """
+
+        # Fully qualified dataset name, i.e. "projects/.../locations/.../datasets/12345"
+        valid_name = extract_fields_from_resource_name(
+            resource_name=dataset_name, resource_noun="datasets"
+        )
+
+        # Partial dataset name (i.e. "12345") with known project and location
+        if (
+            not valid_name
+            and validate_id(dataset_name)
+            and (project or initializer.global_config.project)
+            and (location or initializer.global_config.location)
+        ):
+            dataset_name = DatasetServiceClient.dataset_path(
+                project=project or initializer.global_config.project,
+                location=location or initializer.global_config.location,
+                dataset=dataset_name,
+            )
+
+        # Invalid dataset_name parameter
+        elif not valid_name:
+            raise ValueError("Please provide a valid dataset name or ID")
+
+        super().__init__(project=project, location=location, credentials=credentials)
+        self._gca_resource = self.api_client.get_dataset(name=dataset_name)
+
+    @classmethod
+    def create(
+        cls,
+        display_name: str,
+        metadata_schema_uri: str,
+        metadata: Sequence[Tuple[str, str]] = (),
+        source: Optional[Union[str, List[str]]] = None,
```

Comment:
Elevate source and import_schema_uri in the argument order:

ie.

```python
def create(cls, display_name, source, import_schema_uri...)
```

Reply:
Unfortunately, that elevates default/optional arguments above `metadata_schema_uri` which does not have a default. I'll elevate source and import_schema_uri as such:

```
    def create(
        cls,
        display_name: str,
        metadata_schema_uri: str,
        source: Optional[Sequence[str]] = None,
        import_schema_uri: Optional[str] = None,
        ...
    )
```